crates/net/network/src/manager.rs

```diff
@@ -865,6 +865,44 @@ where
         }
     }
 
+    /// Returns [`PeerInfo`] for all connected peers
+    fn get_peer_infos(&self) -> Vec<PeerInfo> {
+        let peer_manager = self.swarm.state().peers();
+        let mut peers = Vec::with_capacity(peer_manager.num_known_peers());
```

Comment:
this returns the number of peers in the peer set wich is larger than connected peers. this should allocated based on active sessions

Reply:
this will allocate more capacity then necessary because known_peers > active

you can use

```rust
        self.swarm
            .sessions()
            .active_sessions()
            .into_iter()
            .filter_map(|(peer_id, session)| {
                let (record, kind) = self.swarm.state().peers().peer_by_id(*peer_id)?;
                Some(session.peer_info(&record, kind))
            })
            .collect()

```

Reply:
thanks for the advice, I previously thought of `know_peers < active`, don't expect `know_peers > active`, but in which case is `known_peers > active`, the connection is established, but the session not starting?